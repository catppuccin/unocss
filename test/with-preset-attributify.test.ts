import { test, expect } from 'vitest';
import { createGenerator, presetMini, presetAttributify } from 'unocss';

import { extendCatppuccin } from '../dist';

test('presetMini + presetAttributify without setting variant', async () => {
  const uno = createGenerator({
    presets: [
      presetMini({
        dark: 'media',
      }),
      presetAttributify(),
    ],
    theme: {
      colors: extendCatppuccin(),
    },
  });

  const fixture = `
<body bg-latte-base dark:bg-mocha-base>
  <a text-mocha-violet hover:text-mocha-yellow>Hello world!</a>
  <input outline-frappe-base disabled:outline-mocha-base></input>
  <img border-macchiato-red/50 hover:border-macchiato-red></img>
</body>
`;

  const { css } = await uno.generate(fixture, { preflights: false });

  expect(css).toMatchSnapshot();
});

test('presetMini + presetAttributify with default mocha variant', async () => {
  const uno = createGenerator({
    presets: [
      presetMini({
        dark: 'media',
      }),
      presetAttributify(),
    ],
    theme: {
      colors: extendCatppuccin({
        variant: 'mocha',
      }),
    },
  });

  const fixture = `
  <body bg-base dark:bg-base>
    <a text-violet hover:text-yellow>Hello world!</a>
    <input outline-base disabled:outline-base></input>
    <img border-red/50 hover:border-red></img>
  </body>
  `;

  const { css } = await uno.generate(fixture, { preflights: false });

  expect(css).toMatchSnapshot();
});
