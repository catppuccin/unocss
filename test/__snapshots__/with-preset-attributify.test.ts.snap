// Vitest Snapshot v1

exports[`presetMini + presetAttributify with default mocha variant 1`] = `
"/* layer: default */
.bg-base,
[bg-base=\\"\\"]{--un-bg-opacity:1;background-color:rgba(30,30,46,var(--un-bg-opacity));}
.border-red\\\\/50{border-color:rgba(243,139,168,0.5);}
[border-red=\\"\\"],
[hover\\\\:border-red=\\"\\"]:hover{--un-border-opacity:1;border-color:rgba(243,139,168,var(--un-border-opacity));}
.text-violet,
[text-violet=\\"\\"]{--un-text-opacity:1;color:rgba(167,139,250,var(--un-text-opacity));}
[hover\\\\:text-yellow=\\"\\"]:hover{--un-text-opacity:1;color:rgba(249,226,175,var(--un-text-opacity));}
.outline-base,
[disabled\\\\:outline-base=\\"\\"]:disabled,
[outline-base=\\"\\"]{--un-outline-color-opacity:1;outline-color:rgba(30,30,46,var(--un-outline-color-opacity));}
@media (prefers-color-scheme: dark){
[dark\\\\:bg-base=\\"\\"]{--un-bg-opacity:1;background-color:rgba(30,30,46,var(--un-bg-opacity));}
}"
`;

exports[`presetMini + presetAttributify without setting variant 1`] = `
"/* layer: default */
.bg-latte-base,
[bg-latte-base=\\"\\"]{--un-bg-opacity:1;background-color:rgba(239,241,245,var(--un-bg-opacity));}
.border-macchiato-red\\\\/50{border-color:rgba(237,135,150,0.5);}
[border-macchiato-red=\\"\\"],
[hover\\\\:border-macchiato-red=\\"\\"]:hover{--un-border-opacity:1;border-color:rgba(237,135,150,var(--un-border-opacity));}
[hover\\\\:text-mocha-yellow=\\"\\"]:hover{--un-text-opacity:1;color:rgba(249,226,175,var(--un-text-opacity));}
.outline-frappe-base,
[outline-frappe-base=\\"\\"]{--un-outline-color-opacity:1;outline-color:rgba(48,52,70,var(--un-outline-color-opacity));}
[disabled\\\\:outline-mocha-base=\\"\\"]:disabled{--un-outline-color-opacity:1;outline-color:rgba(30,30,46,var(--un-outline-color-opacity));}
@media (prefers-color-scheme: dark){
[dark\\\\:bg-mocha-base=\\"\\"]{--un-bg-opacity:1;background-color:rgba(30,30,46,var(--un-bg-opacity));}
}"
`;
